### ISSUE TITLE
`@plugin` documentation

### ISSUE THREAD

**Issue by calvinjuarez** [less/less-docs#328]
This is a general call for documentation for the use of `@plugin` at http://lesscss.org/usage/#plugins.  (See less/less.js#2416, and pull requests less/less.js#2479 & less/less.js#2522).

**Comment by matthew-dean** [comment]
Yeah, this needs to happen. It's probably Less's coolest new feature that has yet to be known about.

**Comment by Ciantic** [comment]
Is this feature working already? I really need to use it.

**Comment by calvinjuarez** [comment]
I'm with @Ciantic, there needs to be an easier way to get to Less from the plugin.

**Comment by thorn0** [comment]
Actually, it works like this:

**test.js**

```js
functions.add('myfunction', function() {
  return new tree.Color('ff0000');
});
```

The variables `functions` and `tree` are available because [that's how it is implemented](https://github.com/less/less.js/blob/60a5c3bd1f6807615d017a5019031da47e1f480d/lib/less/plugins/function-importer.js): 
```js
new Function("functions", "tree", "fileInfo", contents);
```

**Comment by matthew-dean** [comment]
@thorn0 `less` is also available within a `@plugin` call in 3.x.

https://github.com/less/less.js/blob/3.x/lib/less/environment/abstract-plugin-loader.js#L65
